### PR TITLE
GCE service account sharding

### PR DESCRIPTION
**Pull request by meatballhat**
Further details available in https://github.com/travis-ci/team-build-env-linux/issues/77.

This has already been applied and tested in staging.

### CODE REVIEW

File: modules/gce_worker/main.tf [hunk 1]
```diff
@@ -23,6 +31,18 @@ variable "worker_image" {
   default = "ubuntu-os-cloud/ubuntu-1804-lts"
 }
 
+variable "worker_service_accounts_count_com" {
+  default = 4
```
Comment (Lyoness) [inline modules/gce_worker/main.tf]: any reason for 4 as default?  Just a number to begin with?

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
This looks good to me!

We could also consider creating a first class concept of a shard and then have one instance group per shard, with a dedicated service account per group. That would make the worker process start more predictable, and allow us to use instance service accounts again. :)

It would also pave the way for moving more stuff to a single project, and allow us to run `worker-deploy` in smaller batches.

To be clear, all of the above is by no means a blocker, happy to move forward as-is and think about those things Later(tm). 👍

**Comment by meatballhat** [comment]
I am extremely on board with a first class shard concept :saxophone: :bear:

File: modules/gce_worker/main.tf (hunk 1 above)
Comment (meatballhat) [reply modules/gce_worker/main.tf]: My thinking was that it's the default `length(var.zones)` :man_shrugging:

**Pull request merged**